src/components/avatar-volume-controls.js

```diff
@@ -41,20 +41,28 @@ AFRAME.registerComponent("avatar-volume-controls", {
     this.changeVolumeBy(-1 * step);
   },
 
-  update() {
-    const audioOutputMode = window.APP.store.state.preferences.audioOutputMode;
-    if (
-      this.networkedAudioSource &&
-      this.networkedAudioSource.sound &&
-      (audioOutputMode === undefined || audioOutputMode === "panner")
-    ) {
-      const globalVoiceVolume =
-        window.APP.store.state.preferences.globalVoiceVolume !== undefined
-          ? window.APP.store.state.preferences.globalVoiceVolume
-          : 100;
-      this.networkedAudioSource.sound.gain.gain.value = (globalVoiceVolume / 100) * this.data.volume;
-    }
-  },
+  update: (function() {
+    const positionA = new THREE.Vector3();
+    const positionB = new THREE.Vector3();
+    return function update() {
+      const audio = this.avatarAudioSource && this.avatarAudioSource.el.getObject3D(this.avatarAudioSource.attrName);
+      if (!audio) {
+        return;
+      }
+
+      const { audioOutputMode, globalVoiceVolume } = window.APP.store.state.preferences;
+      const volumeModifier = (globalVoiceVolume !== undefined ? globalVoiceVolume : 100) / 100;
+      let gain = volumeModifier * this.data.volume;
+      if (audioOutputMode === "audio") {
+        this.avatarAudioSource.el.object3D.getWorldPosition(positionA);
+        this.el.sceneEl.camera.getWorldPosition(positionB);
+        const squaredDistance = positionA.distanceToSquared(positionB);
+        gain = gain * Math.min(1, 10 / Math.max(1, squaredDistance));
```

Comment:
This shouldn't be in `update` - it was intentionally originally added to `tick` otherwise the volume attenuation would only happen `update` happens to be called (very noticeably less frequently).

Reply:
Oh nvm I see you're calling update on every `tick` now